### ISSUE TITLE
WP3 Validation of process metadata: test_process_metadata_functions() should not test non-functional parameters within the schema parameter when -m "not longrunning" is added

### ISSUE THREAD

**Issue by Jay-Gohil** [Open-EO/openeo-test-suite#54]
test_process_metadata_functions() from src/openeo_test_suite/tests/processes/metadata/test_process_metadata.py, check values within functional and non-functional parameters. For example, this function also tests for description, which is present in the schema parameter. Like`
"id": "array_append",
"parameters": [
    {
          "description": "Value to append to the array.",
          "name": "value",
          "schema": {
            "description": "Any data type is allowed."
    }
    }
`
\
This should not happen if `-m "not optional"` is added to the command.

**Comment by soxofaan** [comment]
non-functional parameters are skipped with `-m "not optional"`. The `longrunning` marker has nothing to do with process metadata.

See the docs in readme about "WP3 Validation of process metadata":
> Non-functional tests concern descriptions and other metadata of processes that have no impact on the actual behavior of the process, failures in these tests should be taken as warnings, but don't necessarily need to be fixed. These can be skipped by adding `-m "not optional"` to the pytest command.

**Comment by Jay-Gohil** [comment]
There's was a mistake in last sentence. I have made the edit.

**Comment by soxofaan** [comment]
ok, I understand now, you are talking about non-functional fields in the schema of process parameters. The parameter's schema  is indeed being compared as a whole currently:

https://github.com/Open-EO/openeo-test-suite/blob/5b9727d6b6b8a05672bcdd1aafd0a289f32a472a/src/openeo_test_suite/tests/processes/metadata/test_process_metadata.py#L79-L80


Do you have some concrete examples of differing parameter schema's?

**Comment by m-mohr** [comment]
A couple of examples where even schemas are meant to be updated by implementations:

- https://processes.openeo.org/#resample_spatial (method, more specifically the enum and description)
- https://processes.openeo.org/#ard_normalized_radar_backscatter and https://processes.openeo.org/#sar_backscatter (options parameter)
- https://processes.openeo.org/#ard_surface_reflectance (atmospheric_correction_options and cloud_detection_options)
- ...
